src/state/tokens/hooks.ts

```
@@ -283,3 +287,58 @@ export function useTokenTransactions(address: string): Transaction[] | undefined
   // return data
   return transactions
 }
+
+export function useWhitelistTokenByChain(): WhiteListTokenMapByChain {
+  return useSelector<AppState>((state) => state.tokens.mapWhitelistToken) as WhiteListTokenMapByChain
+}
+
+function formatWhitelistToken(tokens: WrappedToken[]): WhiteListTokenMap {
+  return tokens.reduce<WhiteListTokenMap>((tokenMap: WhiteListTokenMap, token: WrappedToken) => {
+    const address: string = token.address || ''
+    if (address) tokenMap[address] = token
+    return tokenMap
+  }, {}) as WhiteListTokenMap
+}
+
+export const fetchAllTokenWhiteList = async (): Promise<WhiteListTokenMapByChain> => {
+  try {
+    const chainPromises = SUPPORTED_NETWORKS.map((chainId) => getWhitelistToken(chainId))
+    const data = await Promise.allSettled(chainPromises)
+    const TokenMap: WhiteListTokenMapByChain = {}
+    data.forEach((e, index) => {
+      TokenMap[SUPPORTED_NETWORKS[index]] = e.status === 'fulfilled' ? e.value : {}
+    })
+    return TokenMap
+  } catch (error) {
+    return {} as WhiteListTokenMapByChain
```

Should not ignore the error here.

---

okay, added console.error